File: scriptedmain.rb
#!/usr/bin/env ruby

def main
	puts "Directory: #{Dir.pwd}"

	puts "Program: #{$0}"

	puts "Number of Args: #{ARGV.length}"

	ARGV.each { |arg| puts "Arg: #{arg}" }
end

if __FILE__ == $0
	main
end

File: scriptedmain.m
#!/usr/bin/env octave -qf

function scriptedmain()
endfunction

function main()
	printf("Directory: %s\n", pwd());
	printf("Program: %s\n", program_name());
	printf("Number of Args: %d\n", nargin);

	args = argv();

	for i = 1:nargin
		printf("Arg: %s\n", args{i});
	endfor

endfunction

main();